### PR TITLE
TASK: Pagination of history

### PR DESCRIPTION
**Pull request by patrickreck**
This adds simple pagination to the history module.

### CODE REVIEW

File: TYPO3.Neos/Classes/TYPO3/Neos/Controller/Module/Management/HistoryController.php [hunk 1]
```diff
@@ -34,18 +34,32 @@ class HistoryController extends AbstractModuleController
...
      * @return void
      */
-    public function indexAction()
+    public function indexAction($offset = 0, $limit = 10)
     {
-        $events = $this->eventRepository->findRelevantEvents()->toArray();
+        $events = $this->eventRepository->findRelevantEvents($offset, $limit + 1)->toArray();
+
+        if (count($events) == $limit + 1) {
+            $events = array_slice($events, 0, 10);
```
Comment (daniellienert) [inline TYPO3.Neos/Classes/TYPO3/Neos/Controller/Module/Management/HistoryController.php]: Shouldn't the "10" be $limit?

Comment (patrickreck) [inline TYPO3.Neos/Classes/TYPO3/Neos/Controller/Module/Management/HistoryController.php]: Absolutely

**Comment by kitsunet** [comment]
@patrickreck Would you be able to merge master or rebase so that there are no conflicts anymore? Would like to get this in. Great stuff!

**Comment by kdambekalns** [comment]
The styling cleanup is probably nice, but first thing I noticed was this before/after glitch, affecting overall alignment:
before
after

**Comment by kdambekalns** [comment]
Loading more history works fine, though.

**Comment by kdambekalns** [comment]
Like this, I suggest some improvements in https://github.com/patrickreck/neos-development-collection/pull/2

**Comment by kdambekalns** [comment]
:+1: from me, then ;)

**Comment by aertmann** [comment]
:-1: styling broken 
image

**Comment by kitsunet** [comment]
@aertmann Maybe some browser cache issue? I cannot reproduce the problem, just tested this change. So :+1: from me.

**Comment by aertmann** [comment]
right seems to work now, sorry :+1:

File: TYPO3.Neos/Resources/Private/Templates/Module/Management/History/Index.html [hunk 2]
```diff
@@ -35,4 +36,47 @@
 			<strong>{neos:backend.translate(id: descriptionTranslationId, source: 'Modules', arguments: descriptionTranslationArguments) -> f:format.raw()}</strong>
 		</div>
 	</div>
-</f:section>
\ No newline at end of file
+</f:section>
+
+<f:if condition="{nextPage}">
+  <div class="loadMore" data-neos-history-nextpage="{nextPage}">
+	  <button>Load more</button>
```
Comment (aertmann) [inline TYPO3.Neos/Resources/Private/Templates/Module/Management/History/Index.html]: needs to a translatable label

File: TYPO3.Neos/Classes/TYPO3/Neos/Controller/Module/Management/HistoryController.php [hunk 3]
```diff
@@ -34,18 +34,32 @@ class HistoryController extends AbstractModuleController
...
+     * @param integer $limit
      * @return void
      */
-    public function indexAction()
+    public function indexAction($offset = 0, $limit = 10)
     {
-        $events = $this->eventRepository->findRelevantEvents()->toArray();
+        $events = $this->eventRepository->findRelevantEvents($offset, $limit + 1)->toArray();
+
+        if (count($events) == $limit + 1) {
```
Comment (aertmann) [inline TYPO3.Neos/Classes/TYPO3/Neos/Controller/Module/Management/HistoryController.php]: ===

**Comment by aertmann** [comment]
found one issue though :-1:

**Comment by kitsunet** [comment]
Will open a follow up for the two mentioned problems.

**Pull request merged**